File: server/app/model/database/user-stats.ts
import { Prop, Schema } from '@nestjs/mongoose';

export type PlayerHistoryDocument = UserStats & Document;

@Schema()
export class UserStats {
    @Prop({ required: true })
    gamesPlayed: number;

    @Prop({ required: true })
    gamesWon: number;

    @Prop({ required: true })
    meanDifferencesFoundPerGame: number;

    @Prop({ required: true })
    timeMeanPerGame: number;
}

File: light_client/lib/general-classes/user-stats.dart
class UserStats {
  final int gamesPlayed;
  final int gamesWon;
  final double meanDifferencesFoundPerGame;
  final double timeMeanPerGame;

  const UserStats(
      {required this.gamesPlayed,
      required this.gamesWon,
      required this.meanDifferencesFoundPerGame,
      required this.timeMeanPerGame});

  UserStats.fromJson(Map<String, dynamic> json)
      : gamesPlayed = json["gamesPlayed"],
        gamesWon = json["gamesWon"],
        meanDifferencesFoundPerGame = json["meanDifferencesFoundPerGame"],
        timeMeanPerGame = json["timeMeanPerGame"];

  Map<String, dynamic> toJson() => {
        'gamesPlayed': gamesPlayed,
        'gamesWon': gamesWon,
        'meanDifferencesFoundPerGame': meanDifferencesFoundPerGame,
        'timeMeanPerGame': timeMeanPerGame,
      };
}
